PKHeX/Legality/Structures/EvolutionTree.cs

```diff
@@ -416,6 +416,25 @@ public IEnumerable<DexLevel> getExplicitLineage(PKM pkm, int lvl, bool skipCheck
                         continue;
 
                     oneValid = true;
+                    if (evo.Level == 0 && !evo.RequiresLevelUp) //Evolutions like elemental stones, trade, etc
+                    {
+                        dl.Last().MinLevel = 1;
+                    }
+                    else if(evo.Level ==0) //Evolutions like frienship, picho -> pikachu, eeve -> umbreon, etc
+                    {
+                        dl.Last().MinLevel = 2;
+                        if (dl.Count() > 1 && !dl.First().RequiresLvlUp)
```

Comment:
Nit pick: It's better to use `List<T>.Count` instead of the Linq `Count()` extension.  The extension is optimized so that the difference is negligible in this case (since `dl` is a List), so I'd only recommend changing it if you have to change something else.